tests/sles4sap/saptune/mr_test.pm

```diff
@@ -95,6 +95,160 @@ sub test_bsc1152598 {
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#1_4";
+    assert_script_run "! saptune note rename ${note} newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#1_4";
+
+    ### Renaming a created Note (without override/with override + not applied/applied)
+
+    assert_script_run "rm -f /etc/saptune/extra/* /etc/saptune/override/*";
+
+    # (applied, no override)
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_1";
+    assert_script_run "EDITOR=/bin/echo saptune note create testnote";
+    assert_script_run "saptune note apply testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_2";
+    assert_script_run "saptune note rename testnote newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_2";
+
+    # (applied, override)
+    assert_script_run "EDITOR=/bin/echo saptune note customise testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_3";
+    assert_script_run "saptune note rename testnote newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_3";
+
+    # (not applied, override)
+    assert_script_run "saptune note revert testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_4";
+    assert_script_run "echo y | saptune note rename testnote newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_4";
+    assert_script_run "echo y | saptune note rename testnote newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_5";
+
+    # (not-applied, no override)
+    assert_script_run "rm /etc/saptune/override/newnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_6";
+    assert_script_run "echo y | saptune note rename newnote testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_6";
+    assert_script_run "echo y | saptune note rename newnote testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#2_7";
+
+    ### Renaming a created Note to an existing note (not applied)
+
+    assert_script_run "rm -f /etc/saptune/extra/* /etc/saptune/override/*";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#3_1";
+    assert_script_run "EDITOR=/bin/echo saptune note create testnote";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#3_2";
+    assert_script_run "! saptune note rename testnote ${note}";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#3_2";
+
+    ### Renaming a non-existent Note
+
+    assert_script_run "rm -f /etc/saptune/extra/* /etc/saptune/override/*";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#4_1";
+    assert_script_run "! saptune note rename 999999 999999";
+    assert_script_run "mr_test verify ${dir}/testpattern_saptune-rename#4_1";
+}
```

Comment:
Wouldn't it be better to have all these blocks as their own functions? I think it can improve readability.

Reply:
Splitted the `test_rename_delete` function into `test_rename` & `test_delete`.